plugins/inputs/monit/monit.go

```diff
@@ -0,0 +1,393 @@
+	} `xml:"load"`
+	CPU struct {
+		User   float64 `xml:"user"`
+		System float64 `xml:"system"`
+		Wait   float64 `xml:"wait"`
+	} `xml:"cpu"`
+	Memory struct {
+		Percent  float64 `xml:"percent"`
+		Kilobyte int64   `xml:"kilobyte"`
+	} `xml:"memory"`
+	Swap struct {
+		Percent  float64 `xml:"percent"`
+		Kilobyte float64 `xml:"kilobyte"`
+	} `xml:"swap"`
+}
+
+type Monit struct {
+	Address           string
+	BasicAuthUsername string
+	BasicAuthPassword string
+	parser            parsers.Parser
+	client            *http.Client
+	tls.ClientConfig
+	Timeout internal.Duration `toml:"timeout"`
+}
+
+func (m *Monit) Description() string {
+	return "Read metrics and status information about processes managed by Monit"
+}
+
+var sampleConfig = `
+  ## Monit
+  address = "http://127.0.0.1:2812"
+
+  ## Username and Password for Monit
+  basic_auth_username = ""
+  basic_auth_password = ""
+
+  ## Amount of time allowed to complete the HTTP request
+  # timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+func (m *Monit) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *Monit) SetParser(parser parsers.Parser) {
+	m.parser = parser
+}
+
+func (m *Monit) Init() error {
+	tlsCfg, err := m.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	m.client = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: m.Timeout.Duration,
+	}
+	return nil
+}
+
+func (m *Monit) Gather(acc telegraf.Accumulator) error {
+
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/_status?format=xml", m.Address), nil)
+	if err != nil {
+		return err
+	}
+	if len(m.BasicAuthUsername) > 0 || len(m.BasicAuthPassword) > 0 {
+		req.SetBasicAuth(m.BasicAuthUsername, m.BasicAuthPassword)
+	}
+
+	resp, err := m.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	var status Status
+	decoder := xml.NewDecoder(resp.Body)
+	decoder.CharsetReader = charset.NewReaderLabel
+	if err := decoder.Decode(&status); err != nil {
+		return fmt.Errorf("Cannot parse input with error: %v\n", err)
+	}
+
+	tags := map[string]string{"address": m.Address, "version": status.Server.Version, "hostname": status.Server.LocalHostname, "platform_name": status.Platform.Name}
+
+	for _, service := range status.Services {
+		fields := make(map[string]interface{})
+		fields["status_code"] = service.Status
+		fields["status"] = serviceStatus(service)
+		fields["monitoring_status_code"] = service.MonitoringStatus
+		fields["monitoring_status"] = monitoringStatus(service)
+		fields["monitoring_mode_status"] = monitoringMode(service)
+		fields["monitoring_mode_code"] = service.MonitorMode
+		tags["service"] = service.Name
+		tags["service_type"] = service.Type
+		if service.Type == "0" {
```

Comment:
These integer types are difficult to understand when reading the code.  Let's define an enum for each of these and parse them in the XML.

Reply:
Done. Please check